### PR TITLE
Feature/as 3293 add sso

### PR DESCRIPTION
**Pull request by swdpcomputing**
Things to note:

Unit tests have been written, although it's difficult to tell whether this will actually 'work' until implemented.

The main.njk layout is included as part of this ticket, however, this should probably be moved from all repos into a service-common/share-views package.

mocks.js is included, but should also be moved into a service-common/mocks or /test-utils package.

### CODE REVIEW

File: services/sso/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "parserOptions": {
+    "ecmaVersion": 2020,
+    "sourceType": "module"
+  },
+  "extends": ["airbnb-base", "prettier"],
+  "plugins": ["jest", "prettier"],
+  "root": true,
+  "ignorePatterns": ["node_modules/**", "src/public/javascript/**", "dist/**", "webpack.**"],
```
Comment (shepda) [inline services/sso/.eslintrc:9]: Doesn't look like a dist dir is being created or webpack is being used so do we need to include those here?

File: services/sso/.gitignore [hunk 2]
```diff
@@ -0,0 +1,180 @@
+# Created by .ignore support plugin (hsz.mobi)
```
Comment (shepda) [inline services/sso/.gitignore:1]: There are a lot of file types included here that we don't use so do we need to include them?

File: services/sso/src/controllers/your-planning-appeal.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const { VIEW } = require("../lib/views");
```
Comment (shepda) [inline services/sso/src/controllers/your-planning-appeal.js:1]: As this is a common service it should ideally be more generic and not include controllers for specific services as it shouldn't need to know about the services that consume it. Does this need to be included here?

File: services/sso/src/routes/authenticationRouter.test.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  });
+
+  it("should call passport with a mock strategy", () => {
+    authenticationRouter(req, res, mockStrategyId);
+    expect(passport.authenticate).toHaveBeenCalledWith(mockStrategyId, {
+      failureRedirect: "/",
+    });
+  });
+
+  it("should define the expected route with an existant user", () => {
```
Comment (shepda) [inline services/sso/src/routes/authenticationRouter.test.js:27]: I believe existant should actually be spelt existent :)

File: services/sso/src/routes/authenticationRouter.test.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    req.user = { mockUser: "mock-user" };
+
+    authenticationRouter(req, res, mockStrategyId);
+    expect(res.redirect).toHaveBeenCalledWith("/logged-in");
+  });
+
+  it("should define the expected route with an non-existant user", () => {
```
Comment (shepda) [inline services/sso/src/routes/authenticationRouter.test.js:34]: I believe non-existant should actually be spelt non-existent :)

File: services/sso/.eslintrc (hunk 1 above)
Comment (swdpcomputing) [reply services/sso/.eslintrc:9]: This is just copying and pasting from the template. I think it's probably not a good use of time editing these templates to fit every common service. Happy to discuss.

File: services/sso/.gitignore (hunk 2 above)
Comment (swdpcomputing) [reply services/sso/.gitignore:1]: This is just copying and pasting from the template. I think it's probably not a good use of time editing these templates to fit every common service. Happy to discuss.

File: services/sso/src/routes/authenticationRouter.test.js (hunk 4 above)
Comment (swdpcomputing) [reply services/sso/src/routes/authenticationRouter.test.js:27]: Edited, thank you.

File: services/sso/src/routes/authenticationRouter.test.js (hunk 5 above)
Comment (swdpcomputing) [reply services/sso/src/routes/authenticationRouter.test.js:34]: Edited, thank you.

File: services/sso/.eslintrc (hunk 1 above)
Comment (swdpcomputing) [reply services/sso/.eslintrc:9]: Edited.

File: services/sso/.gitignore (hunk 2 above)
Comment (swdpcomputing) [reply services/sso/.gitignore:1]: Edited.

**Review by shepda — APPROVED** [review]
LGTM

**Pull request merged**